test/test_optim.py

```diff
@@ -539,6 +539,23 @@ def test_adam(self):
             with self.assertRaisesRegex(ValueError, "Invalid weight_decay value: -1"):
                 optimizer(None, lr=1e-2, weight_decay=-1)
 
+    def test_adam_complex(self):
+        for optimizer in [optim.Adam,optim_mt.Adam]:
+            self._test_complex_optimizer(
+                lambda param: optimizer([param], lr=0.001)
+            )
+            self._test_complex_optimizer(
+                lambda param: optimizer([param], lr=0.001, betas=(0.3,0.555))
+            )
+            self._test_complex_optimizer(
+                lambda param: optimizer([param], lr=0.001, betas=(0.3,0.555), eps = 1e-6 )
+            )
+            self._test_complex_optimizer(
+                lambda param: optimizer([param], lr=0.001,  betas=(0.3,0.555), eps = 1e-6, weight_decay=1 )
+            )
+            self._test_complex_optimizer(
+                lambda param: optimizer([param], lr=0.001,  betas=(0.3,0.555), eps = 1e-6, weight_decay=1, amsgrad=True)
+            )
     # Test whether variance parameter is always real
     def test_complex_adam_variance(self):
```

Comment:
It is the `test_complex_adam_variance` function report an error. Do we still need it?

Reply:
I guess we don't strictly need it but this should still be true no?

Reply:
I don't think it is true now since we compute the variance independently for the real part and imaginary part.